### DISCUSSION TITLE
Accessing Data in .vuepress/config.ts from somewhere else

### DISCUSSION THREAD

**Discussion by derogate** [vuepress/core#704] [Q&A] [answered]
For example, how do I access [global variables defined in .vuepress/config.ts](https://v2.vuepress.vuejs.org/reference/plugin-api.html#define) from .vue files or clientAppEnhance.ts? I don't find those global variables in getCurrentInstance() in .vue files or from app argument exposed in clientAppEnhance.ts. Althought I defined something in config.ts, it is not globally defined outside of it

**Comment by Mister-Hope** [comment, marked as answer]
I am not getting your point.

If you want to access something in node side from client. Use define options in plugin API, then you can access them in files directly like local variable. They will be replaced by the thing you defined while processing files.

**Reply by derogate** [reply]
Just realized that my issue is Typescript-related & variables using define options in plugin API is already accessible globally in globalThis context. Need to setup global.d.ts manually for IDE to properly detect the type.
